src/test/java/com/spotify/confidence/ConfidenceTest.java

```diff
@@ -0,0 +1,275 @@
+
+  @Test
+  void getValue() {
+    final Confidence confidence = Confidence.create(fakeEngine, fakeFlagResolverClient);
+    final Integer value = confidence.getValue("flag.prop-E", 20);
+    assertEquals(50, value);
+
+    final FlagEvaluation<Integer> evaluation = confidence.getEvaluation("flag.prop-E", 20);
+
+    assertEquals(50, evaluation.getValue());
+    assertEquals("flags/flag/variants/var-A", evaluation.getVariant());
+    assertEquals("RESOLVE_REASON_MATCH", evaluation.getReason());
+    assertTrue(evaluation.getErrorType().isEmpty());
+    assertTrue(evaluation.getErrorMessage().isEmpty());
+  }
+
+  @Test
+  void getFullValue() {
+    final Confidence confidence = Confidence.create(fakeEngine, fakeFlagResolverClient);
+    final Struct value = confidence.getValue("flag", Struct.EMPTY);
+    // This mocked struct is defined in ResolverClientTestUtils
+    final Struct expected =
+        Struct.builder()
+            .set("prop-A", ConfidenceValue.of(false))
+            .set(
+                "prop-B",
+                Struct.builder()
+                    .set("prop-C", ConfidenceValue.of("str-val"))
+                    .set("prop-D", ConfidenceValue.of(5.3))
+                    .build())
+            .set("prop-E", ConfidenceValue.of(50))
+            .set(
+                "prop-F",
+                ConfidenceValue.List.of(List.of(ConfidenceValue.of("a"), ConfidenceValue.of("b"))))
+            .set(
+                "prop-G",
+                Struct.builder()
+                    .set("prop-H", ConfidenceValue.NULL_VALUE)
+                    .set("prop-I", ConfidenceValue.NULL_VALUE)
+                    .build())
+            .build();
+    assertEquals(expected, value);
+
+    final FlagEvaluation<Struct> evaluation = confidence.getEvaluation("flag", Struct.EMPTY);
+
+    assertEquals(expected, evaluation.getValue());
+    assertEquals("flags/flag/variants/var-A", evaluation.getVariant());
+    assertEquals("RESOLVE_REASON_MATCH", evaluation.getReason());
+    assertTrue(evaluation.getErrorType().isEmpty());
+    assertTrue(evaluation.getErrorMessage().isEmpty());
+  }
+
+  @Test
+  void getValueIncompatibleType() {
+    final Confidence confidence = Confidence.create(fakeEngine, fakeFlagResolverClient);
+    final FlagEvaluation<String> evaluation = confidence.getEvaluation("flag.prop-E", "test");
+    assertEquals("test", evaluation.getValue());
+    assertEquals("", evaluation.getVariant());
+    assertEquals("ERROR", evaluation.getReason());
+    assertEquals(ErrorType.INVALID_VALUE_TYPE, evaluation.getErrorType().get());
+    assertEquals(
+        "Default type class java.lang.String, but value of "
+            + "type class com.spotify.confidence.ConfidenceValue$Integer",
+        evaluation.getErrorMessage().get());
+  }
+
+  @Test
+  void getValueUnsupportedType() {
+    final Confidence confidence = Confidence.create(fakeEngine, fakeFlagResolverClient);
+    final FlagEvaluation<Date> evaluation =
```

Comment:
nit: would ready easier if it was not imported but actually used
```suggestion
    final FlagEvaluation<java.sql.Date> evaluation =
```